charts/add-ons/prometheus/values.yaml

```diff
@@ -0,0 +1,54 @@
+image: prom/prometheus:v2.15.2
+args:
+  storage.tsdb.path: /data
+  storage.tsdb.retention.time: 6h
+  config.file: /etc/prometheus/prometheus.yml
+  log.level: debug
+globalConfig:
+  scrape_interval: 10s
+  scrape_timeout: 10s
+  evaluation_interval: 10s
+scrapeConfigs:
```

Comment:
What's the reasoning behind putting some scrape configs here, and others in `prometheus.yaml`?

Reply:
Oh the ones that depend on other values are placed in the template... You could place them in the original place in the main `values.yaml` and instead of using `{{.Values.global.namespace}}` declare a yaml anchor for `namespace` and use `*namespace`.
As for `{{default .Values.global.proxyContainerName "linkerd-proxy" .Values.global.proxyContainerName}}` that custom value is such an obscure setting that we might as well get rid of it. It's not even mentioned in the chart's README.md nor in `values.yaml`.

Reply:
Did not quite understand your comment about anchoring namespace in `values.yaml`
But that would mean we would have to hardcode Linkerd and the global.namespace is not propagated till here right? when users configure their own namespace using `global.namespace`.

Reply:
Just going to move the scrape-configs into templates as they are not conflicting, only other fields like args are i.e when they are overwritten.